R/scale-date.r

```diff
@@ -125,7 +125,14 @@ scale_datetime <- function(aesthetics, trans,
     minor_breaks <- date_breaks(date_minor_breaks)
   }
   if (!is.waive(date_labels)) {
-    labels <- date_format(date_labels)
+    # uses context from breaks to pass timezone
+    labels <- function(breaks) {
+      if ("POSIXt" %in% class(breaks)) {
```

Comment:
I think we should make `tz` an explicit parameter, defaulting it to NULL (meaning use the time zone from the data).

You have a couple of style problems:

* Use `inherits()` not `%in% class(x)`
* `ifelse()` is for vectorised comparisons, here `tz %||% "UTC"` is adequate
* You're missing spaces around `=`